internal/server/receiver_handlers.go

```diff
@@ -367,26 +362,79 @@ func (s *ReceiverServer) annotate(ctx context.Context, resource apiv1.CrossNames
 
 	group, version := getGroupVersion(apiVersion)
 
+	if resource.Name == "*" {
+		logger.Info(fmt.Sprintf("annotate resources by matchLabel for kind '%s' in '%s'",
+			resource.Kind, namespace), "matchLabels", resource.MatchLabels)
+
+		var resources metav1.PartialObjectMetadataList
+		resources.SetGroupVersionKind(schema.GroupVersionKind{
+			Group:   group,
+			Kind:    resource.Kind,
+			Version: version,
+		})
+
+		if err := s.kubeClient.List(ctx, &resources,
+			client.InNamespace(namespace),
+			client.MatchingLabels(resource.MatchLabels),
+		); err != nil {
+			return err
```

Comment:
As we're not logging the resource details in the caller, anymore, we need to add these here:
```suggestion
			return fmt.Errorf("failed listing resources in namespace %q by matching labels %q: %w", namespace, err)
```